.github/workflows/freertos_demos.yml

```diff
@@ -66,10 +68,44 @@ jobs:
           sudo apt-get -y update
           sudo apt-get -y install build-essential
 
-      - name: Build Posix_GCC Demo
+      - name: Build Posix_GCC Full Demo
+        id: build-posix-gcc-demo
         shell: bash
         working-directory: FreeRTOS/Demo/Posix_GCC
-        run: make -j
+        run: |
+          make -j
+
+      - name: Run and monitor Posix_GCC Full Demo
+        if: success()
+        working-directory: FreeRTOS/Demo/Posix_GCC
+        run: |
+          timeout 5m build/posix_demo | tee output.txt
+          if grep -q "OK: No errors" output.txt; then
+            echo -e "\nSUCCESS - Found SUCCESS string in output."
+          else
+            echo -e "\nFAILURE - SUCCESS string not found in output."
+            exit 1
+          fi
+
+      - name: Build Posix_GCC Blinky Demo
+        if: success() || failure() && steps.build-posix-gcc-demo.outcome == 'success'
```

Comment:
```suggestion
        id: build-posix-gcc-blinky-demo
        if: success() || failure()
```

Reply:
adding this id will allow you to differentiate between the two make steps. success() || failure() will make this run despite previous steps failing, but will still allow you to cancel the flow. You could add an ID to the install gcc step and add it as a dependency as well but probably not needed